Automattic/Gravatar-SDK-Android
# Fix: Loading state of the profile component is not matching the design

Closes #195

### Description

The loading state for all the profile cards did not correctly represent what the design created. This PR aims to bring them closer to the "what we want" state. It won't be 100% correct yet, because I have found other issues (https://github.com/Automattic/Gravatar-SDK-Android/issues/199) and I didn't want to push all the changes in one PR. 

This one focuses on the Skeleton object width and padding. 

Here's the comparison using designs from Figma and screenshots from our test showing the changes:

| Design | Before - After |
|---|---|
| <img width="256" alt="Screenshot 2024-07-01 at 10 50 41" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/12660801/7c50f207-5353-49e3-a243-962de3d4b83f"> | <img width="1214" alt="Screenshot 2024-07-01 at 10 50 17" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/12660801/13aabe98-616e-4751-97ac-cba5f1f114c3"> |
| <img width="259" alt="Screenshot 2024-07-01 at 10 50 38" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/12660801/cb7b4946-b45a-4e38-b551-e4cdb07ff2e4"> | <img width="1213" alt="Screenshot 2024-07-01 at 10 50 31" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/12660801/c6e1e4e8-334f-4389-b523-1e47f94a5223"> |
| <img width="232" alt="Screenshot 2024-07-01 at 10 50 52" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/12660801/6d308cc8-c2d0-48a1-b300-01c7100a0bf1"> | <img width="1203" alt="Screenshot 2024-07-01 at 10 49 52" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/12660801/86e635b3-0bdf-4d08-aff5-d309be4e7a6d"> |
| <img width="232" alt="Screenshot 2024-07-01 at 10 50 57" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/12660801/08c5cb45-2e98-4b5d-9e65-29737760a585"> | <img width="1208" alt="Screenshot 2024-07-01 at 10 48 59" src="https://github.com/Automattic/Gravatar-SDK-Android/assets/12660801/235dba39-4559-4953-9909-ac1ef354d651"> |

**Why and what**

To achieve that I had to add a separate `skeletonModifier` to control the size and styling of the skeleton object ([as per this](https://developer.android.com/reference/kotlin/androidx/compose/ui/Modifier#:~:text=Composables%20that%20accept%20modifiers%20to%20be%20applied%20to%20a%20specific%20subcomponent%20foo%20should%20name%20the%20parameter%20fooModifier%20and%20follow%20the%20same%20guidelines%20above%20for%20default%20values%20and%20behavior.%20Subcomponent%20modifiers%20should%20be%20grouped%20together%20and%20follow%20the%20parent%20composable%27s%20modifier.%20For%20example%3A)). It is required because of how all the components, that take in the `ComponentState`, are designed. There's no single parent composable - based on the state we create different top composables so sharing the modifier is not an option.

Alternatively, we could add that missing top composable, but this would still require us to pass in the modifier conditionally as those would differ depending on the `ComponentState`.  

### Testing Steps
1. Run the demo app
2. Go to the Profile tab
3. Tap the `Skeleton` button
4. Confirm the UI looks properly and is closer to the designs
